components/script/task_source/unshipped_port_message_queue.rs

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
Is this implementation necessary? I wasn't able to infer it from the documentation (https://html.spec.whatwg.org/multipage/web-messaging.html#unshipped-port-message-queue) and I'm not sure I understand what's a "virtual" task source.

I was hoping to use the PR as an excuse to get guidance with this :).

Thanks!